### PR TITLE
feat(completion): add ComplMatchIns highlight currently inserted completion

### PR DESCRIPTION
**Pull request by glepnir**
Added ComplMatchIns highlight group to highlight the currently inserted completion item.

image

### CODE REVIEW

**Comment by benknoble** [comment]
I personally think it’s a neat idea. I assume there’s a way to undo the effect (say, by unsetting or relinking a highlight group) for folks who don’t want it?

**Comment by glepnir** [comment]
yes we can set PmeuMatchIns higlight like normal in default. if you want enable just give it a special fg color.

**Comment by glepnir** [comment]
I updated the current behavior. Leader will not be affected by PmenuMatchIns. Only the text filled by the completion is affected ,  `ello` is inserted with `bold and red` .. any thoughts ? or affected the whole inserted word `hello`

image

**Comment by glepnir** [comment]
emm this is Ci error no matter PR

**Comment by glepnir** [comment]
@zeertzjq  @chrisbra  Could you guys PTAL

**Comment by zeertzjq** [comment]
I think this should be independent of the popup menu. (E.g the highlight group name shouldn't contain `Pmenu`.) Insert mode completion can work without popup menu.

File: src/drawline.c [hunk 1]
```diff
@@ -3995,6 +4004,7 @@ win_line(
 	    else
 		ScreenCols[wlv.off] = -1;
 
+
```
Comment (zeertzjq) [inline src/drawline.c]: ?

**Comment by glepnir** [comment]
aha yes and name can also renew i guess.

**Comment by glepnir** [comment]
> I think this should be independent of the popup menu

Yes. If there is only one result it will be inserted directly. So does it need to be highlighted? Currently using `pum_visible` to check on win_line, if so then the current condition may not be sufficient

**Comment by glepnir** [comment]
emm yes... so currently pum_visible to work only when pum is displayed. does it need to be changed here?

**Comment by chrisbra** [comment]
How long does this stay? I suppose only during completion? In my opinion looks a bit too much eye-candy.

**Comment by glepnir** [comment]
yes only when completion any other edit or change will make it restore to normal attribute.

**Comment by glepnir** [comment]
adjusted the logic. now only give the text inserted by the currently selected completion item a highlight group

File: runtime/doc/syntax.txt [hunk 2]
```diff
@@ -5857,6 +5857,9 @@ PmenuThumb	Popup menu: Thumb of the scrollbar.
 PmenuMatch	Popup menu: Matched text in normal item.
 							*hl-PmenuMatchSel*
 PmenuMatchSel	Popup menu: Matched text in selected item.
+
+CompMatchIns	Matched text of the currently inserted completion.
```
Comment (zeertzjq) [inline runtime/doc/syntax.txt]: `Compl` is more common than `Comp`

Comment (glepnir) [reply runtime/doc/syntax.txt]: neat

**Pull request closed without merging**